### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by sbshah97**
#1 partially fulfilled!

* CONTRIBUTING.md file added
* README.md file updated!

@chinmaydd  @mohitreddy1996 @adeepkit01 kindly review!

### CODE REVIEW

**Review by adeepkit01 — COMMENTED** [review]
I think the flow from git clone to git push is misplaced, as the aim of contributing document is to setup guidelines for the coding standards for this repository, which is the guidelines the code in the pull request should follow, how that pull request is created is UpTo them, but as I said it's a personal opinion

**Comment by sbshah97** [comment]
@adeepkit01 why not have both in the coding guidelines! The essential idea behind how to push the code structure was to prevent people from not disturbing the current directory structure. For example we don't want people to put all their Java Files in the root folder instead of taking them in a separate folder called Java!

File: README.md [hunk 1]
```diff
@@ -12,15 +12,12 @@ The project aims to build a library which will provide a cache implementation th
...
 * Remove an existing item from the cache.
 
-## Contributing
+## Contributing 
 
-We have members with varying experiences in different programming languages! Want to contribute? Amazing :) Please follow guidelines as below:
+* If you are looking to contribute code, kindly look at the [CONTRIBUTING.md](CONTRIBUTING.md) file for information as to how to contribute to the repository!
 
-* If you are implementing the above logic in a programming language X, make sure you name your git branch as `X` and raise a PR for the same. If it already exists, try making it better!
-* Benchmark your programs against a test suite. We will be adding the module soon. Comparing different implementations hence becomes easier.
```
Comment (mohitreddy1996) [inline README.md]: Let this point be there. I will be adding few test cases later as mentioned in this line. But those test cases would be independent of the programming language as we can't write testing modules for every programming language being implemented by others. 
So a generic testing module will be set up. Seeing which they will have to create test module in that specific programming language.

**Comment by mohitreddy1996** [comment]
@salman-bhai please check a comment i have put in the README file. 
@adeepkit01 @chinmaydd I think having a contribution document separately is fine. I have seen many organizations having a contribution document separately. So I feel we can apply the proposed changes once you both review it.

**Comment by sbshah97** [comment]
@mohitreddy1996 cannot find the comment that you've mentioned! Did you put it as a commit or did you submit a review for the Pull Request!

**Comment by sbshah97** [comment]
Changes made as suggested by @mohitreddy1996 ! Kindly squash commits if it's an all-clear!

**Pull request merged**